zk-keygen/src/main.rs

```diff
@@ -158,6 +238,81 @@ fn do_main(matches: &ArgMatches) -> Result<(), Box<dyn error::Error>> {
                 }
             }
         }
+        ("pubkey", matches) => {
+            let key_type = KeyType::get_key_type(matches);
+
+            let mut path = dirs_next::home_dir().expect("home directory");
+            let path = if matches.is_present("keypair") {
+                matches.value_of("keypair").unwrap()
+            } else {
+                path.extend([".config", "solana", key_type.default_file_name()]);
+                path.to_str().unwrap()
+            };
+
+            // wrap the logic inside a match statement in case more keys are supported in the
+            // future
+            match key_type {
+                KeyType::ElGamal => {
+                    let elgamal_pubkey =
+                        elgamal_keypair_from_path(matches, path, "pubkey recovery", false)?.public;
+
+                    if matches.is_present("outfile") {
+                        let outfile = matches.value_of("outfile").unwrap();
+                        check_for_overwrite(outfile, matches)?;
+                        elgamal_pubkey.write_to_file(outfile)?;
+                    } else {
+                        println!("{elgamal_pubkey}");
+                    }
+                }
+                _ => unreachable!(),
+            }
+        }
+        ("recover", matches) => {
+            let key_type = KeyType::get_key_type(matches);
+
+            let mut path = dirs_next::home_dir().expect("home directory");
+            let outfile = if matches.is_present("outfile") {
+                matches.value_of("outfile").unwrap()
+            } else {
+                path.extend([".config", "solana", key_type.default_file_name()]);
+                path.to_str().unwrap()
+            };
+
+            if outfile != STDOUT_OUTFILE_TOKEN {
+                check_for_overwrite(outfile, matches)?;
+            }
+
+            match key_type {
+                KeyType::ElGamal => {
+                    let keypair_name = "recover";
+                    let keypair = if let Some(path) = matches.value_of("prompt_signer") {
+                        elgamal_keypair_from_path(matches, path, keypair_name, true)?
+                    } else {
+                        let skip_validation =
+                            matches.is_present(SKIP_SEED_PHRASE_VALIDATION_ARG.name);
+                        elgamal_keypair_from_seed_phrase(
+                            keypair_name,
+                            skip_validation,
+                            true,
+                            None,
+                            true,
+                        )?
+                    };
+                    output_encodable_key(&keypair, outfile, "recovered ElGamal keypair")?;
+                }
+                KeyType::Aes128 => {
+                    let key_name = "recover";
```

Comment:
Move outside match, since it's the same for both?

Reply:
Yeah, I originally made it separate since variable names were different, but I think the variable names are less important here. I just renamed the variable `name` and moved it outside the match.